proposals/continuations/Explainer.md

```diff
@@ -0,0 +1,1277 @@
+ .                     .           .                     .
+                                   .                     .
+```
+
+The stack pointer moves to the top of stack 2, and thus execution
+continues on stack 2.
+
+
+```ioke
+      (stack 1)                         (stack 2)
+       (suspended)
+ |---------------------|
+ |                     |
+ | ...                 |
+ |                     |                 (active)
+ |                     |       ----|---------------------|
+ | $h1                 |<-----/   >| $f()                |
+ .                     .           |                     |
+ .                     .           .                     .
+ .                     .           .                     .
+                                   .                     .
+```
+
+As execution continues on stack 2 it may eventually perform a
+`suspend`, which will cause another transfer of
+control. Supposing it invokes `suspend` with some `$e` handled
+by `$h1`, then stack 2 will transfer control back to stack 1.
+
+
+```ioke
+      (stack 1)                         (stack 2)
+       (suspended)
+ |---------------------|
+ |                     |
+ | ...                 |
+ |                     |                 (active)
+ |                     |       ----|---------------------|
+ | $h1                 |<-----/    | ...                 |
+ .                     .          >| suspend $e     |
+ .                     .           .                     .
+ .                     .           .                     .
+                                   .                     .
+```
+
+The suspension reverses the parent-child link again, and leaves behind
+a "hole" on stack 2, that can be filled by an invocation of
+`resume`.
+
+```ioke
+      (stack 1)                         (stack 2)
+       (suspended)
+ |---------------------|
+ |                     |
+ | ...                 |
+ |                     |                 (active)
+ |                     |       --->|---------------------|
+>| $h1                 |------/    | ...                 |
+ .                     .           | [ ]                 |
+ .                     .           .                     .
+ .                     .           .                     .
+                                   .                     .
+```
+
+
+
+## Design Considerations and Extensions
+
+### Memory Management
+
+The current proposal does not require a cycle-detecting garbage
+collector as the linearity of continuations guarantees that there are
+no cycles in continuation objects. In theory, we could do without any
+automated memory management at all if we took seriously the idea that
+failure to use a continuation constitutes a bug in the producer. In
+practice, we expect that for many systems the only feasible way to
+ensure that continuations are always used (or at least deallocated) is
+to use some form of automated memory management, such as a reference
+counting.
```

Comment:
```suggestion
no cycles in continuation objects. To prevent space leaks from unconsumed
continuations, implementations will generally need some form of automated
memory management for them. Due to their acyclicity, a reference counting scheme is sufficient.
```

Reply:
something I wanted to say has been deleted here... I'll try rewording this